frontend/packages/ceph-storage-plugin/src/__mocks__/volume-snapshot-data.tsx

```diff
@@ -0,0 +1,88 @@
+import { FirehoseResourcesResult, FirehoseResult } from '@console/internal/components/utils';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export const snapshotData = {
+  items: [
+    {
+      apiVersion: 'snapshot.storage.k8s.io/v1beta1',
+      kind: 'VolumeSnapshot',
+      metadata: {
+        name: 'fakeSnapshot',
+        namespace: 'fakens',
+      },
+      spec: {
+        source: {
+          persistentVolumeClaimName: 'fake-pvc',
+        },
+        volumeSnapshotClassName: 'fake-snc',
+      },
+      status: {
+        readyToUse: true,
+        restoreSize: '1Gi',
+      },
+    },
+  ],
+  breadcrumbs: [
+    {
+      name: 'Persistent Volume Claims',
+      path: `/k8s/ns/fakens/persistentvolumeclaims`,
+    },
+    {
+      name: 'Snapshot Details',
+      path: '/k8s/ns/default/kind/example',
+    },
+  ],
+  fakeModel: {
+    abbr: 'fk',
+    kind: 'fake',
+    label: 'Fake',
+    labelPlural: 'Fakes',
+    plural: 'fakes',
+    apiVersion: 'v1',
+    apiGroup: 'fakeapi',
+  },
+};
+
+export const newSnapshotData = {
+  items: [
+    {
+      apiVersion: 'snapshot.storage.k8s.io/v1beta1',
+      kind: 'VolumeSnapshot',
+      metadata: {
+        name: 'fakeSnapshot',
+        namespace: 'fakens',
+      },
+      spec: {
+        source: {
+          persistentVolumeClaimName: 'fake-pvc',
+        },
+      },
+    },
+  ],
+};
+
+export const pvcData: FirehoseResourcesResult = {
+  data: ({
+    apiVersion: 'v1',
+    kind: 'PersistentVolumeClaim',
+    metadata: {
+      name: 'fake-pvc',
+      namespace: 'fakens',
+    },
+    spec: {
+      resources: {
+        requests: {
+          storage: '1Gi',
+        },
+      },
+      storageClassName: 'fake-scn',
+    },
+    status: {
+      phase: 'Bound',
+      accessModes: ['ReadWriteOnce'],
+      capacity: {
+        storage: '1Gi',
+      },
+    },
+  } as unknown) as FirehoseResult<K8sResourceCommon | K8sResourceCommon[]>,
```

Comment:
I'd suggest to skip typing of `pvcData` variable here and force-cast it to `FirehoseResult< .. >` in the test that imports & uses it.

More generally, typing to `any` or `unknown` and then to a specific type should be avoided.